bitcoin/src/blockdata/script/tests.rs

```diff
@@ -311,23 +312,23 @@ fn scriptint_round_trip() {
         -((1 << 31) - 1),
     ];
     for &i in test_vectors.iter() {
-        assert_eq!(Ok(i), read_scriptint(&build_scriptint(i)));
-        assert_eq!(Ok(-i), read_scriptint(&build_scriptint(-i)));
+        assert_eq!(Ok(i), PushBytes::read_scriptint(&build_scriptint(i)));
+        assert_eq!(Ok(-i), PushBytes::read_scriptint(&build_scriptint(-i)));
         assert_eq!(Ok(i), read_scriptint_non_minimal(&build_scriptint(i)));
         assert_eq!(Ok(-i), read_scriptint_non_minimal(&build_scriptint(-i)));
     }
-    assert!(read_scriptint(&build_scriptint(1 << 31)).is_err());
-    assert!(read_scriptint(&build_scriptint(-(1 << 31))).is_err());
+    assert!(PushBytes::read_scriptint(&build_scriptint(1 << 31)).is_err());
+    assert!(PushBytes::read_scriptint(&build_scriptint(-(1 << 31))).is_err());
     assert!(read_scriptint_non_minimal(&build_scriptint(1 << 31)).is_err());
     assert!(read_scriptint_non_minimal(&build_scriptint(-(1 << 31))).is_err());
 }
 
 #[test]
 fn non_minimal_scriptints() {
-    assert_eq!(read_scriptint(&[0x80, 0x00]), Ok(0x80));
-    assert_eq!(read_scriptint(&[0xff, 0x00]), Ok(0xff));
-    assert_eq!(read_scriptint(&[0x8f, 0x00, 0x00]), Err(Error::NonMinimalPush));
-    assert_eq!(read_scriptint(&[0x7f, 0x00]), Err(Error::NonMinimalPush));
+    assert_eq!(PushBytes::read_scriptint(&[0x80, 0x00]), Ok(0x80));
+    assert_eq!(PushBytes::read_scriptint(&[0xff, 0x00]), Ok(0xff));
+    assert_eq!(PushBytes::read_scriptint(&[0x8f, 0x00, 0x00]), Err(Error::NonMinimalPush));
```

Comment:
> This must take &self there's no point of this change otherwise.

If `read_scriptint` function would take `&self` then  how we can make this thing run as `read_scriptint` would then need   the struct instance but we don't have any public method to create `PushBytes` instance other than having `PushBytes::empty()` method .

Reply:
> we don't have any public method to create `PushBytes` instance

There's `AsRef` for arrays with known length (like in this case they are 2-byte arrays) and `TryFrom` for slices with unknown length.